src/components/App.jsx

```diff
@@ -34,17 +34,16 @@ export default class Table extends React.Component {
     }
 
     /**
-    * Function to sort the data in the state by a key.
-    * @param key The key name in the JSON data. Can be one of 'id', 'author', 'title', 'year', 'citations'
-    **/
+     * Function to sort the data in the state by a key.
+     * @param key The key name in the JSON data. Can be one of 'id', 'author', 'title', 'year', 'citations'
+     **/
     sortJsonDataByKey(key) {
-      var tempData = this.state.jsonData;
-      var sortOrder = this.state.sortOrder[key];
+      let tempData = this.state.jsonData;
```

Comment:
I believe this can still be `const` because the object that it is referencing is not changing.

Reply:
It does change. It changes because sort function is called on that variable. That is why the state is set to tempData after the sort function.